### ISSUE TITLE
Website download file

### ISSUE THREAD

**Issue by greenchili** [h5bp/html5-boilerplate#2147]
Hi, this is a feature request I suppose. The zip file download from the home page is a bit of a 'zip bomb'. I think it would be better to unzip into a contained folder as this can be quite annoying.

**Comment by roblarsen** [comment]
Hi, can you further describe your issue with the zip file? I can guess what you mean, but I would only be guessing.

**Comment by greenchili** [comment]
Download the zip file from the site:
The file from your main site:

https://github.com/h5bp/html5-boilerplate/releases/download/v7.2.0/html5-boilerplate_v7.2.0.zip

It unzips the files directly to the destination directory rather than a self contained folder.

**Comment by coliff** [comment]
Extracting any ZIP by default extracts to a folder with the same name as the ZIP. If I right-click extract it (by default) gets extracted to a '**html5-boilerplate_v7.2.0**' folder on Windows 10. I just tested on MacOS 10.14.6 too and it does the same. So it works well and as expected.

If it were to be in an additional folder it's actually more hassle and annoying as its an additional folder to dig down to.

**Comment by greenchili** [comment]
I am unzipping on linux cli using the standard unzip command:
unzip --help                      
UnZip 6.00 of 20 April 2009, by Debian. Original by Info-ZIP.

**Comment by webhat** [comment]
I can confirm, it unzips into the current working directory.

> I am unzipping on linux cli using the standard unzip command:
> unzip --help
> UnZip 6.00 of 20 April 2009, by Debian. Original by Info-ZIP.

**Comment by coliff** [comment]
ah ok. Maybe UnZip for Linux needs an update so it behaves like Windows and MacOS then :-)

Really, there isn't anything that can realistically be done here. The GitHub ZIP release just takes the files from the repo and makes it available as a ZIP as is so the only way round that would be to put everything in a folder first, but that'd be quite odd (I've never seen another GitHub repo do that) - and it'd annoying for all Windows and MacOS users.

**Comment by roblarsen** [comment]
To be precise the download file that Github produces is different than the release zip file. The release zip file is intentionally structured this way so that the unzipped folder on Mac or Windows is nice and tidy. I am sorry I've never tried the download file on linux. For now, the dowload file of latest or the tag generated by Github (literally browsing to the tag and clicking the "clone or download" button) won't be as clean (it will just be the project folder) but will be contained inside a folder.

**Comment by greenchili** [comment]
OK if you feel this a non issue then please close.

**Comment by eclectic-coding** [comment]
For what it's worth, the ZIP file downloaded from Github (using Ubuntu ARCHIVE  extracts into a new directory named `HTML5-BOILERPLATE-MASTER`. However, when the file is downloaded from the web site there is not root directory to extract into, the user has to direct to a new empty directory. I see what @greenchili is referring to. 
IMHO, these archives should behave the same.

**Comment by roblarsen** [comment]
Some notes: 

The zip from Github is different than the released zip file. I have control over one and no control over the other. 

The file we include with the release is the dist folder.  I _intentionally_ zip it up so that (on Mac and Windows at least) it behaves in the way we're describing here. You perform the default unzip action and the folder output is the contents of the dist folder. We've since learned that's not optimal on Linux. Since it's a legit issue I'm leaving this open.  If there's a way to make the release better AND be nice to linux users than that would be something I would do for the next release. 

The zip file from Github is just a snapshot of the state of the _repo_ that you're viewing. I don't have any control over how that is zipped up.

**Comment by jms1989** [comment]
file-roller on linux unzips to a folder named the same as the zip file when done from the file manager. 

you can just do:

```
mkdir html5-boilerplate
unzip html5-boilerplate*.zip -d html5-boilerplate
```

**Comment by roblarsen** [comment]
That might be the best we can do, document it in that way.

**Comment by Kevin-hope** [comment]
Hello everyone this is China good evening

**Comment by acluther** [comment]
Has the documentation been updated to reflect the ZIP file behavior? I'd be glad to take it on if not.

**Comment by roblarsen** [comment]
@acluther it has not. You can definitely take it on. I'm just queuing up the project for a new release so your timing is excellent.

**Comment by acluther** [comment]
@roblarsen great, I'll take it on this evening.

**Comment by roblarsen** [comment]
@acluther if you're still interested I'd love to include this update in the upcoming release

**Comment by Diraneh** [comment]
Bonjour, 

Il faut d'abord compresse  votre donnée dans un fichier en zip et puis  <p> Téléchargez mon super <a href="nondevotredonnee.html.zip">fichier</a></p>.

**Comment by acluther** [comment]
> @acluther if you're still interested I'd love to include this update in the upcoming release

I had every good intention of getting on it but other things ended up taking priority. I apologize but would like to be involved in the future.

**Comment by roblarsen** [comment]
closed via #2169 branch
